### PR TITLE
Wire Signup and Login Use cases

### PR DESCRIPTION
**Pull request by cristhianescobar**

### CODE REVIEW

File: android/app-project-newm/src/main/java/io/projectnewm/MainActivity.kt [hunk 1]
```diff
@@ -25,16 +26,45 @@ class MainActivity : ComponentActivity() {
 }
 
 @Composable
-fun WelcomeToNewm() {
+fun WelcomeToNewm(signupViewModel: SignupViewModel = org.koin.androidx.compose.get()) {
```
Comment (wlara) [inline android/app-project-newm/src/main/java/io/projectnewm/MainActivity.kt:29]: what do you think about using `koinViewModel()` here, like:
```
fun WelcomeToNewm(signupViewModel: SignupViewModel = koinViewModel())
```

File: android/feature-login/src/main/java/io/projectnewm/feature/login/screen/LoginScreen.kt [hunk 2]
```diff
@@ -36,29 +35,24 @@ internal const val TAG_LOGIN_SCREEN = "TAG_LOGIN_SCREEN"
 
 @Composable
 fun LoginScreen(
-    onSignInSubmitted: (email: String, password: String) -> Unit,
-    scrollState: ScrollState = rememberScrollState()
+    onUserLoggedIn: () -> Unit,
+    onSignupClick: () -> Unit,
+    viewModel: LoginViewModel = org.koin.androidx.compose.get()
```
Comment (wlara) [inline android/feature-login/src/main/java/io/projectnewm/feature/login/screen/LoginScreen.kt:40]: Same here. Maybe use `koinViewModel()`.

**Review by wlara — APPROVED** [review]

**Comment by martyu** [comment]
@cristhianescobar is this building for you?  I get "Unresolved reference" errors for LoginStatus and isValid.

File: shared/src/commonMain/kotlin/io.projectnewm.shared/login/models/LoginResponse.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package io.projectnewm.shared.login.models
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class LoginResponse(
```
Comment (martyu) [inline shared/src/commonMain/kotlin/io.projectnewm.shared/login/models/LoginResponse.kt:7]: I don't think LoginResponse needs to be returned by LoginUseCase does it?  I thought accessToken and refreshToken would be stored in KMM.  I don't think the clients would need to access them directly eh?

File: shared/src/commonMain/kotlin/io.projectnewm.shared/login/models/LoginResponse.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @SerialName("accessToken")
+    val accessToken: String,
+    @SerialName("refreshToken")
+    val refreshToken: String
+)
+
+sealed class LoginStatus {
```
Comment (martyu) [inline shared/src/commonMain/kotlin/io.projectnewm.shared/login/models/LoginResponse.kt:14]: Is there a way we can just throw these errors?  Or if we decide to share view models, we can just set the error there and the client view's can show them.

**Review by martyu — APPROVED** [review]

**Pull request merged**